### ISSUE TITLE
Source custom label is sometimes deleted by itself

### ISSUE THREAD

**Issue by flogiston1** [foundryvtt/dnd5e#2627]
After migration to system version 2.4.0 old source fields were migrated to the new source objects "custom label". Sometimes after opening an item with such source data, and not changing the source data in any way it gets deleted between closing the item and reopening it again.

Not sure if the item needs to be updated in some other way for this to happen or what actions causes this behaviour to occur.

Tried just now to open a background item (from a compendium) that had not yet been opened after world migration, changed some stuff but not the source field, closed item, refreshed browser window, opened the item again and the source field was empty.

**Comment by arbron** [comment]
Haven't been able to duplicate this myself. The migration is explicitly checking as to whether source is an object before it does anything, and the field being present on the sheet when custom label is used should ensure that any changes are propagated properly into the data model.

Just to verify, were you using the core system sheets or a 3rd party sheet?

**Comment by Fyorl** [comment]
One other thing to check: Does it only happen to items opened from a compendium? And does it only happen after some time as passed? In which case, it might be related to the compendium cache expiring.

**Comment by flogiston1** [comment]
This was with core system sheets. In the example I gave above not much time passed as I specifically tried to reproduce this.
If I can find out more details about when/how this occurs, I'll post an update.

**Comment by flogiston1** [comment]
Maybe some new info:

This was my own shared compendium that I tried this with. This time I tried with a subclass item. The item had sourcedata only on custom label, which I did not touch in any way (the picture is from another item after this test).
![image](https://github.com/foundryvtt/dnd5e/assets/151408500/d8b595fc-c4bd-489f-9dc6-d8f1b63280ae)

I added advancement data (choice or itemgrant advancement) to the subclass (I did the same with the background item before but trait advancement), closed the card, refreshed the page (F5), opened the same item again and the source data was gone. 

This did not happen when I only edited the item's description so it seems to be connected to advancement.

I managed to reproduce this multiple times.

This only seems to happen if the source data has not been modified after migration to new system. To test this, I changed the source custom label field before trying again, and this time it did not get deleted.

This was on a node.js installation. The modules active are just my own shared compendiums.
![image](https://github.com/foundryvtt/dnd5e/assets/151408500/018c7b29-6dae-419b-ac6c-17f6a7778f64)

So this is not an issue if I update the source on all my stuff first, before adding advancement. Still might be unpleasant if a lot of data is accidentally deleted.
